src/instructlab/model/serve_backend.py

```diff
@@ -23,6 +28,55 @@ def signal_handler(num_signal, __):
     sys.exit(0)
 
 
+def get_gpu_count() -> int:
+    # Third Party
+    import torch
+
+    if torch.cuda.is_available():
+        return torch.cuda.device_count()
+    return 0
+
+
+def check_gpu_count(gpus: int, available_gpus: int) -> bool:
+    return gpus > available_gpus
+
+
+def check_tensor_parallel_size(
+    vllm_args: List, available_gpus: int
+) -> Tuple[bool, int]:
+    """
+    Checks if --tensor-parallel-size argument is valid.
+
+    Returns:
+        - (True, tensor_parallel_size) if tensor-parallel-size is invalid (greater than available GPUs).
+        - (False, 0) if --tensor-parallel-size is not found or invalid.
```

Comment:
This reads as True or False could be returned as invalid.  Which I think is true from the code below.  Is that desired?

Reply:
Thank you for your remind, I updated the logic.